SDK/MobileCenterAnalytics/VSMC.Analytics.Windows.Shared/Analytics.cs

```diff
@@ -154,6 +171,71 @@ private ISessionTracker CreateSessionTracker(IChannelGroup channelGroup, IChanne
             return _sessionTrackerFactory?.CreateSessionTracker(channelGroup, channel) ?? new SessionTracker(channelGroup, channel);
         }
 
+        /// <summary>
+        /// Validates name.
+        /// </summary>
+        /// <param name="name">Log name to validate.</param>
+        /// <param name="logType">Log type.</param>
+        /// <returns><c>true</c> if validation succeeds, otherwise <с>false</с>.</returns>
+        private bool ValidateName(string name, string logType)
+        {
+            if (string.IsNullOrEmpty(name))
+            {
+                MobileCenterLog.Error(LogTag, logType + " name cannot be null or empty.");
+                return false;
+            }
+            if (name.Length > MaxEventNameLength)
+            {
+                MobileCenterLog.Error(LogTag, string.Format("{0} '{1}' : name length cannot be longer than {2} characters.", logType, name, MaxEventNameLength));
+                return false;
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Validates properties.
+        /// </summary>
+        /// <param name="properties">Properties collection to validate.</param>
+        /// <param name="logName">Log name.</param>
+        /// <param name="logType">Log type.</param>
+        /// <returns>Valid properties collection with maximum size of 5</returns>
+        private IDictionary<string, string> ValidateProperties(IDictionary<string, string> properties, string logName, string logType)
+        {
+            if (properties == null)
+                return null;
```

Comment:
>return null; [](start = 16, length = 12)

nitpick: Omitting curly bracket in one-line return statement is error-prone. See microsoft stylecop SA1503 ([http://stylecop.soyuz5.com/sa1503.html](http://stylecop.soyuz5.com/sa1503.html))